src/sinks/gcp/cloud_storage.rs

```diff
@@ -377,10 +390,14 @@ impl RequestSettings {
             .acl
             .map(|acl| HeaderValue::from_str(&to_string(acl)).unwrap());
         let content_type = HeaderValue::from_str(encoder.content_type()).unwrap();
-        let content_encoding = config
-            .compression
-            .content_encoding()
-            .map(|ce| HeaderValue::from_str(&to_string(ce)).unwrap());
+        let content_encoding = if config.set_content_encoding {
```

Comment:
Have you looked at how S3 allows configuring content_encoding? They have a `content_encoding` option [1] which can override the content encoding derived from the compression method. I would guess this would probably have much higher changes of being accepted upstream

[1] https://github.com/vectordotdev/vector/blob/master/src/sinks/s3_common/config.rs#L118
[2] https://github.com/vectordotdev/vector/blob/master/src/sinks/s3_common/service.rs#L106

Reply:
I thought about that. However, we want to avoid setting content_encoding altogether, not overriding it with a different value than the one determined by the compression method (gzip).

I can try providing a different content_encoding value ("" or something invalid) and see if Lumberjack can still process these logs.

Reply:
@flaviofcruz I fixed the PR to provide a content_encoding option similar to the S3 sink.
Things worked on Vector Aggregator when content_encoding is set to an empty string.